custom_components/grocy/__init__.py

```diff
@@ -92,36 +115,26 @@
     hass.data[DOMAIN_DATA] = {}
 
     # Get "global" configuration.
-    url = config[DOMAIN].get(CONF_URL)
-    api_key = config[DOMAIN].get(CONF_API_KEY)
-    verify_ssl = config[DOMAIN].get(CONF_VERIFY_SSL)
-    port_number = config[DOMAIN].get(CONF_PORT)
+    url = config_entry.data.get(CONF_URL)
+    api_key = config_entry.data.get(CONF_API_KEY)
+    verify_ssl = config_entry.data.get(CONF_VERIFY_SSL)
+    port_number = config_entry.data.get(CONF_PORT)
+    hash_key = hashlib.md5(api_key.encode('utf-8')).hexdigest()
 
     # Configure the client.
     grocy = Grocy(url, api_key, port_number, verify_ssl)
     hass.data[DOMAIN_DATA]["client"] = GrocyData(hass, grocy)
-
-    # Load platforms
-    for platform in PLATFORMS:
-        # Get platform specific configuration
-        platform_config = config[DOMAIN].get(platform, {})
-
-        # If platform is not enabled, skip.
-        if not platform_config:
-            continue
-
-        for entry in platform_config:
-            entry_config = entry
-
-            # If entry is not enabled, skip.
-            if not entry_config[CONF_ENABLED]:
-                continue
-
-            hass.async_create_task(
-                discovery.async_load_platform(
-                    hass, platform, DOMAIN, entry_config, config
-                )
-            )
+    hass.data[DOMAIN_DATA]["hash_key"] = hash_key
+    hass.data[DOMAIN_DATA]["url"] = url
```

Comment:
The url is probably not needed for unique identification, could you either remove it from unique_id or hash the combination of api key and url?